### DISCUSSION TITLE
centos 5 complie XFreeRDP 2.3.2 ERROR

### DISCUSSION THREAD

**Discussion by muzhen466** [FreeRDP/FreeRDP#7151] [Q&A]
i used the cmd like "cmake -DWITH_FFMPEG=OFF -DWITH_ALSA=OFF -DWITH_XKBFILE=OFF ." and then i used "make",
there is a error message "‘PTHREAD_MUTEX_RECURSIVE’ undefined"
how i can solve this problem?
plz~~~

**Comment by akallabeth** [comment]
By adding a bit more of information.
There is some information at https://github.com/FreeRDP/FreeRDP/wiki/Compilation for a start.
Please always start with a `clean` checkout (and use out of source tree builds), a compile might fail if you tried previously and there are files left over from that.
